urld025f21ac3_mvattipulusu_Innovation2013_tgz-pJ8-com_usps_data_ManifestFileTestJ8/src/com/usps/evs/vo/PicCodeParserManifestV20.java

```diff
@@ -0,0 +1,251 @@
+		} else if ("C07".equals(barcodeConstruct)) {
+			picCodeVo.setPostalRoutingCode(detailVo.getPicCode().substring(0,3));
+			picCodeVo.setZip(detailVo.getPicCode().substring(3, 8));
+			picCodeVo.setApplicationIdentifier(detailVo.getPicCode().substring(8, 10));
+			picCodeVo.setServiceTypeCode(detailVo.getPicCode().substring(10, 13));
+			picCodeVo.setDuns(detailVo.getPicCode().substring(13, 19));
+			picCodeVo.setPackageID(detailVo.getPicCode().substring(19, 29));
+			picCodeVo.setCheckDigit(detailVo.getPicCode().substring(29, 30));
+			picCodeVo.setDunsPkgidDzip(detailVo.getPicCode().substring(8, 30));
+		} else if ("C08".equals(barcodeConstruct)) {
+			picCodeVo.setApplicationIdentifier(detailVo.getPicCode().substring(0, 2));
+			picCodeVo.setServiceTypeCode(detailVo.getPicCode().substring(2, 5));
+			picCodeVo.setDuns(detailVo.getPicCode().substring(5, 11));
+			picCodeVo.setPackageID(detailVo.getPicCode().substring(11, 21));
+			picCodeVo.setCheckDigit(detailVo.getPicCode().substring(21, 22));
+			picCodeVo.setDunsPkgidDzip(detailVo.getPicCode().substring(0,22));
+		} else if ("C09".equals(barcodeConstruct)) {
+			picCodeVo.setApplicationIdentifier(detailVo.getPicCode().substring(0, 2));
+			picCodeVo.setServiceTypeCode(detailVo.getPicCode().substring(2, 5));
+			picCodeVo.setDuns(detailVo.getPicCode().substring(5, 11));
+			picCodeVo.setPackageID(detailVo.getPicCode().substring(11, 25));
+			picCodeVo.setCheckDigit(detailVo.getPicCode().substring(25, 26));
+			picCodeVo.setDunsPkgidDzip(detailVo.getPicCode().substring(0, 26));
+		} else if ("C10".equals(barcodeConstruct)) {
+			picCodeVo.setApplicationIdentifier(detailVo.getPicCode().substring(0, 2));
+			picCodeVo.setServiceTypeCode(detailVo.getPicCode().substring(2, 5));
+			picCodeVo.setDuns(detailVo.getPicCode().substring(5, 14));
+			picCodeVo.setPackageID(detailVo.getPicCode().substring(14, 25));
+			picCodeVo.setCheckDigit(detailVo.getPicCode().substring(25, 26));
+			picCodeVo.setDunsPkgidDzip(detailVo.getPicCode().substring(0, 26));
+		} else if ("L01".equals(barcodeConstruct)) { /* Old format Pic Code in a new format file V1.5 */
+			//Release 32 added check to see of the mailer is sending anything other than 22 digit long PICs
+			if(detailVo.getPicCode().trim().length()==PicCodeParserManifestV15.L01PicCodeLength){
+				picCodeVo.setApplicationIdentifier(detailVo.getPicCode().substring(0, 2));
+				picCodeVo.setServiceTypeCode(detailVo.getPicCode().substring(2, 4));
+				picCodeVo.setDuns(detailVo.getPicCode().substring(4, 13));
+				picCodeVo.setPackageID(detailVo.getPicCode().substring(13, 21));
+				picCodeVo.setCheckDigit(detailVo.getPicCode().substring(21, 22));
+				picCodeVo.setDunsPkgidDzip(detailVo.getPicCode().substring(0, 22));
+			} else {
+				picCodeVo.setDunsPkgidDzip(detailVo.getPicCode().trim());
+				picCodeVo.setApplicationIdentifier(" ");
+				picCodeVo.setPackageID(" ");
+				picCodeVo.setDuns("000000000");
+				picCodeVo.setServiceTypeCode(" ");
+			}
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 134:140, 165:172.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.